### PR TITLE
@radoslawkrzemien/13604 local testing of GitHub actions

### PR DESCRIPTION
**Pull request by radoslawkrzemien**
### Details

Testing framework for running local tests of GitHub Actions workflows to identify issues before pushing them into the repository and testing there

### Fixed Issues

$ https://github.com/Expensify/App/issues/13604 
$ https://github.com/Expensify/App/issues/13604#issuecomment-1405005943

### Tests

1. Run `npm run workflow-test`
2. Verify test run results

- [x] Verify that no errors appear in the JS console

### QA Steps

1. Run `npm run workflow-test`
2. Verify test run results

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

</details>

<details>
<summary>Mobile Web - Chrome</summary>

</details>

<details>
<summary>Mobile Web - Safari</summary>

</details>

<details>
<summary>Desktop</summary>

</details>

<details>
<summary>iOS</summary>

</details>

<details>
<summary>Android</summary>

</details>

### CODE REVIEW

**Comment by radoslawkrzemien** [comment]
This is a clone PR of https://github.com/Expensify/App/pull/15560 with the difference being that these commits are signed

**Comment by dangrous** [comment]
swapping out @rushatgabhane for @sobitneupane since they were on the original PR!

**Comment by sobitneupane** [comment]
It took around 35-40 minutes to complete the test.
Screenshot 2023-07-06 at 15 42 50
Screenshot 2023-07-06 at 15 43 53

**Comment by dangrous** [comment]
So I think the podfile thing is because you're adding [a new node module](https://github.com/Expensify/App/pull/21937/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R147) and so that needs to be updated in the podfile as well. Do we need that? I'm not exactly sure where it's being used.

I'm getting this warning when trying to install pod locally:

```
[!] CocoaPods could not find compatible versions for pod "Airship/MessageCenter":
  In Podfile:
    react-native-airship (from `../node_modules/@ua/react-native-airship`) was resolved to 15.2.6, which depends on
      AirshipFrameworkProxy (= 2.0.8) was resolved to 2.0.8, which depends on
        Airship/MessageCenter (= 16.11.3)

    urbanairship-react-native (from `../node_modules/urbanairship-react-native`) was resolved to 14.6.1, which depends on
      Airship (= 16.10.7) was resolved to 16.10.7, which depends on
        Airship/MessageCenter (= 16.10.7)
```

Otherwise this is looking good to me!

**Comment by dangrous** [comment]
One update - I removed that urbanairship-react-native module and everything still seems to work fine, so that's probably the solution? Do you remember where that came from? Is there a chance it's an old merge conflict and we removed it in another PR?

**Comment by radoslawkrzemien** [comment]
@dangrous Yes, that's definitely some leftover from resolved merge conflicts, as I never that package myself. I'll spend some time today to resolve the new conflicts

**Review by dangrous — DISMISSED** [review]
I think this is a ✅ from me! Definitely would love to see what @sobitneupane and @AndrewGable find too though.

**Comment by sobitneupane** [comment]
Tested it two times. Different test case is failing each time with "Exceeded timeout of 60000 ms for a test."
Screenshot 2023-07-13 at 14 34 07
Screenshot 2023-07-13 at 15 13 21

**Comment by radoslawkrzemien** [comment]
@sobitneupane That's interesting, for me the tests run in less than half the time, with the longest running one clocking just under 25 sec.
Screenshot 2023-07-13 at 11 55 00
I'll increase the timeout to 90 sec., that should sort it out

**Comment by sobitneupane** [comment]
It does the trick. Tests passing now.

Screenshot 2023-07-13 at 16 19 15

**Comment by dangrous** [comment]
Sweet! Whenever you're ready @sobitneupane  (after you've looked through everything) you can go ahead and approve - assuming you don't find anything else - and then hopefully the robot should reassign me (and maybe andrew too?) officially.

**Comment by AndrewGable** [comment]
I am getting some failed tests but it's unclear if it's my set up or what 🤔 Do we have any checks that run so we can trouble shoot the setup? e.g. making sure `act` is installed, etc?

**Comment by dangrous** [comment]
Hm, @AndrewGable you went through the setup instructions in [this](https://github.com/radoslawkrzemien/App/blob/@radoslawkrzemien/13604-local-testing-of-github-actions/workflow_tests/README.md)? I think when similar errors were happening to me I either needed to make sure Docker was actually open, and/or make sure the ACT_BINARY was correctly in my .env.

**Comment by radoslawkrzemien** [comment]
@AndrewGable Please double-check if you have followed all the steps in the workflow_tests/README file. Your issues are very similar to what was happening to @dangrous in the initial stages of the review

**Review by sobitneupane — COMMENTED** [review]
### Screenshots/Videos
<details>

<summary>Test</summary>
Screenshot 2023-07-19 at 21 36 43

</details>
<details>
<summary>Web</summary>

Screenshot 2023-07-19 at 17 30 39

</details>

<details>
<summary>Mobile Web - Chrome</summary>



</details>

<details>
<summary>Mobile Web - Safari</summary>
Screenshot 2023-07-19 at 17 38 07



</details>

<details>
<summary>Desktop</summary>
Screenshot 2023-07-19 at 17 41 45



</details>

<details>
<summary>iOS</summary>



</details>

<details>
<summary>Android</summary>
Screenshot 2023-07-19 at 17 43 25



</details>

**Comment by dangrous** [comment]
> Thanks for the reply, my point is the test and code in this PR should check for an incorrect setup and error when we know it's not setup correctly instead of running and failing.

Ah yeah, that makes sense to me! Would save a lot of time from if we didn't know if it was the test or the setup.

**Comment by dangrous** [comment]
I think we're currently waiting on an update from this comment - https://github.com/Expensify/App/pull/21937#issuecomment-1642209129

**Comment by dangrous** [comment]
> Thanks for the reply, my point is the test and code in this PR should check for an incorrect setup and error when we know it's not setup correctly instead of running and failing.

Bump on [this](https://github.com/Expensify/App/pull/21937#issuecomment-1642209129) when you can, @radoslawkrzemien! Would love to get this out the door soon!

**Comment by radoslawkrzemien** [comment]
Hi @dangrous, I was OOO last week. I will try to add the setup check this week and let you know

**Comment by dangrous** [comment]
Sounds great! Let us know when it's ready for testing again.

**Comment by radoslawkrzemien** [comment]
@dangrous @AndrewGable I have changed the `workflow-test` command to instead of running the tests immediately run a script that checks if 1) docker is installed 2) docker is running 3) act is installed 4) ACT_BINARY is set and only then run the tests. LMK if this is what you had in mind

**Comment by radoslawkrzemien** [comment]
Hi @dangrous , I have modified the script to look for the variable in the `.env` file if it is not set. LMK if this works for you

File: workflow_tests/cherryPick.test.js
Comment (dangrous) [inline workflow_tests/cherryPick.test.js:343]:
```suggestion
    describe('automatic trigger', () => {
```

File: workflow_tests/cherryPick.test.js
Comment (dangrous) [inline workflow_tests/cherryPick.test.js:193]:
```suggestion
                describe('version does not match', () => {
```

File: workflow_tests/cherryPick.test.js
Comment (dangrous) [inline workflow_tests/cherryPick.test.js:292]:
```suggestion
                describe('version does not match', () => {
```

File: workflow_tests/testBuild.test.js
Comment (dangrous) [inline workflow_tests/testBuild.test.js:348]:
```suggestion
        it('executes workflow, without getBranchRef', async () => {
```

File: workflow_tests/testBuild.test.js
Comment (dangrous) [inline workflow_tests/testBuild.test.js:486]:
```suggestion
        it('executes workflow, without getBranchRef', async () => {
```

**Review by dangrous — CHANGES_REQUESTED** [review]
Env check works great! I found a couple last typos and then I think I'm ready to ✅ this!

**Comment by dangrous** [comment]
ack, conflicts. almost there!

**Review by dangrous — DISMISSED** [review]
Okay, I think we're good! @AndrewGable and @sobitneupane let us know if you find anything else.

Once this is merged/deployed we should definitely prep a post calling it out in Slack, and letting people know how and when to use it!

**Comment by dangrous** [comment]
@radoslawkrzemien we've got some conflicts now

Bump @AndrewGable and @sobitneupane if you have time to test - would love to get this out this week!

**Review by sobitneupane — APPROVED** [review]
Changes tests well.

### Screenshots/Videos
<details>
<summary>Tests</summary>
Screenshot 2023-08-14 at 22 51 22

</details>
<details>
<summary>Web</summary>

Screenshot 2023-08-14 at 22 54 21

</details>

<details>
<summary>Mobile Web - Chrome</summary>
Screenshot 2023-08-14 at 23 01 07



</details>

<details>
<summary>Mobile Web - Safari</summary>


Screenshot 2023-08-14 at 22 59 52

</details>

<details>
<summary>Desktop</summary>
Screenshot 2023-08-14 at 22 59 05



</details>

<details>
<summary>iOS</summary>
Screenshot 2023-08-14 at 23 04 09


</details>

<details>
<summary>Android</summary>


</details>

**Comment by dangrous** [comment]
Oh weird it re-requested me. I can approve once those conflicts are fixed, and then I think it's down to you @AndrewGable!

**Review by dangrous — DISMISSED** [review]

**Comment by dangrous** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android / native
    - [ ] Android / Chrome
    - [ ] iOS / native
    - [ ] iOS / Safari
    - [ ] MacOS / Chrome / Safari
    - [ ] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>



</details>

<details>
<summary>Mobile Web - Chrome</summary>



</details>

<details>
<summary>Mobile Web - Safari</summary>



</details>

<details>
<summary>Desktop</summary>



</details>

<details>
<summary>iOS</summary>



</details>

<details>
<summary>Android</summary>



</details>

**Review by AndrewGable — DISMISSED** [review]
Overall looking great!

Questions:
1. I can't remember since we've had this one open for a while, did we agree to using async for some reason specifically for these tests?
2. Should we be running these tests in `test.yml`? 

I think both of these could be handled as a follow up. Overall I think this is great and happy to have @dangrous merge once he is happy 👍

**Comment by radoslawkrzemien** [comment]
I resolved conflicts with `package.json` and your approves got removed 😞 
Please have another look 🙏

**Comment by AndrewGable** [comment]
@radoslawkrzemien - Can you respond to my [questions](https://github.com/Expensify/App/pull/21937#pullrequestreview-1585246217) above? Thanks.

**Comment by radoslawkrzemien** [comment]
@AndrewGable 
1. IIRC this was not discussed previously. However, `act-js` uses async functions anyway
2. Yes, that's one of the ways the feature could be used (as part of the test workflow, along the rest of the tests)

**Comment by AndrewGable** [comment]
1. Understand, but I do not think we want to open up using `async`.. Can we fix this?
2. Sounds good, I think we can leave this for a follow up if we want, but we should add it soon.

**Comment by radoslawkrzemien** [comment]
I think so. Is there a specific reason why using `async-await` is not desired? If so, changing the tests to use `.then()` syntax would be sufficient, or would you like to rewrite the `act-js` lib to not use asynchronous functions at all?

**Comment by AndrewGable** [comment]
Dependencies are fine to use async/await, we just want to use promises directly in our source code.

We've written a bit as to why here: https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#esnext-are-we-allowed-to-use-insert-new-language-feature-why-or-why-not

Using `.then()` is perfect 👍

**Comment by dangrous** [comment]
So I think based on some further internal discussion we can keep async/await here in these files (cc @AndrewGable) - so that eslintrc change is fine. I think it's just the merge conflicts, then (and we might actually be broadening where we allow async so we can broaden that rule, but I haven't heard back yet on where we want it so no change yet. At worst it can be another followup).

**Comment by radoslawkrzemien** [comment]
@dangrous Added `tests` and `.github` to `.eslintrc.js` overrides and resolved conflicts

**Review by dangrous — DISMISSED** [review]
Sweet! We're now in the merge freeze so this will probably have to wait until that's over.... (And so we'll likely need one more commit to fix merge conflicts at the end there.)

But this works for me until then!

Thanks for your patience as this continues on.

**Comment by dangrous** [comment]
Oops, looks like some lint errors too

**Review by dangrous — DISMISSED** [review]
Approving but will hold on a merge through the freeze (unless you feel we can go ahead @AndrewGable, it does seem pretty self-contained so it would probably be fine?)

**Comment by radoslawkrzemien** [comment]
Hey guys, any updates? Is there anything left for me to do, or just waiting now for the merge?

**Comment by dangrous** [comment]
I think we're just waiting for the merge (the freeze should be up at the end of this week) unless @AndrewGable has any final comments!

**Comment by dangrous** [comment]
Hey @radoslawkrzemien can you give a quick summary of the merge conflicts that you updated? It's hard to know exactly what happened since there's so much code haha. Namely:
* It looks like the `Resolve conflicts` commit removes most of the StagingDeployCash steps, is that from something merged in, or intentional for this specific PR?
* This is nitpicky, but I noticed that most of the mock steps created via createMockStep take the form `'Find open StagingDeployCash', 'Finding open StagingDeployCash',` (with the 	`ing` in the second part of details) - can you make sure all the ones you added take that form?

@AndrewGable let us know if you notice anything else, otherwise I think I'll be good to merge after that!

**Review by AndrewGable — APPROVED** [review]

**Comment by radoslawkrzemien** [comment]
Hey @dangrous
I have merged the changes from upstream repository, which also included some changes to the workflows and because of that the tests needed to be updated slightly in order to reflect these changes and keep passing

The StagingDeployCash steps have been removed in `main` and this removal was one of the changes merged, hence I removed the corresponding mocks and assertions as they were no longer needed

As for the `createMockStep` uses, the first value is the name of the step and has to be exactly the same as in the workflow and the second is the message displayed in the logs and returned in the workflow result so it can be set to anything. First I went with the convention to take the step name and modify it into a continuous verb form (as in `Find open StagingDeployCash` -> `Finding open StagingDeployCash`), but after working some more on this task I found it a bit too error prone when creating assertions, and since then I started to use the unmodified step value as the message - its the same way in the script for auto-generating mocks/assertions from the workflow (`preGenerateTest.js`)

LMK if something needs more clarification

**Comment by dangrous** [comment]
Great that all makes sense to me! I think we can merge!

**Review by dangrous — APPROVED** [review]

**Comment by kavimuru** [comment]
@dangrous Could you please help us with 1st step in QA `Run npm run workflow-test` ?

**Comment by dangrous** [comment]
Hey @kavimuru ! I think this won't actually need QA, it only affects dev, so y'all can ignore. (Unless I'm mistaken, @radoslawkrzemien )

**Comment by radoslawkrzemien** [comment]
You are correct @dangrous , it should only affect dev

**Pull request merged**
